### DISCUSSION TITLE
Adopt for EVCC

### DISCUSSION THREAD

**Discussion by lieglw** [PatrikTrestik/homeassistant-solax-http#32] [Ideas]
@PatrikTrestik Can your great work be used for the EVCC custom charger?

**Comment by PatrikTrestik** [comment]
I have no knowledge of EVCC. I'll check how it works.

Actually if EVCC requires some standardized interface and it is documented, I can try to adapt.

**Comment by PatrikTrestik** [comment]
I briefly checked EVCC project integration options.
Because my integration runs inside Home assistant there is no direct use possible in EVCC. It will require some work.

Here are possible options:
- Write EVCC charger template for SolaX API.
There is existing template for SolaX Modbus which can be used as start.
I can provide mapping from Modbus registers to API results and can help with calling API.
I could not do coding here, because I have  no GO skills and also don't have EVCC installed. Search for programmer :)
- Use plugins to integrate through HomeAssistant API
EVCC plugins can call any HTTP. If you are running HA, there is option to call its API and read and write sensors provided by ma integration.
- Similar approach can be used through MQTT.
HA can expose sensors to MQTT (running as addon in HA or separate installation).
EVCC has plugin to integrate MQTT.
- Use JS plugin in EVCC and call SolaX API directly.
It could be possible but again you need to find JavaScript doer with access to SolaX charger for testing.

I can help with mapping and understanding API, but not with actual coding.

**Comment by lieglw** [comment]
Thank You for your help. I have HA running so this would be the best option.
Do you have anything where I can start from?

Thank You

**Reply by PatrikTrestik** [reply]
Check in EVCC community if someone already uses HA API integration. It can show you how to use EVCC HTTP template.

HA API documentation is here: https://developers.home-assistant.io/docs/api/rest/
Read basics and test if you are able to read sensor value by HA REST API.

**Reply by lieglw** [reply]
Thanks, will give a try.

**Reply by empx92** [reply]
> Thanks, will give a try.

Any news on that ;)?

**Reply by lieglw** [reply]
At the moment no time for this.
Think will do it via HA automation.

**Reply by empx92** [reply]
i did it as a HA automation, there are some small issues at the moment which i have to sort out but it is running quite good (maybe i need 1-2 more test days)

i can tell the evc:
-Prioritymode
-max. Pull from Grid

There are 3 Autmations for it:
1.) Priority Mode
2.) Control Charge Status (to Check if the Car is loading without fail, otherwise it will stop the chargemode and try to "reconnect")
3.) The Charging-Control itself

That all is working with Patriks custom component...
@PatrikTrestik i think the charging-control could be integrated more easy in python which i sadly can't program, so if there a interest in furher developing for your custom component i could give you the codes i used to program.

**Reply by PatrikTrestik** [reply]
Hi.
I have week or more time to do home projects.
If you need my help, I'm at hand.
